coveralls/api.py

```diff
@@ -68,6 +64,17 @@ def ensure_token(self):
         if self.config.get('repo_token') or not self._token_required:
             return
 
+        service = self.config.get('service_name') or ''
+        if self._token_required and service.startswith('github'):
+            gh_token = os.environ.get('GITHUB_TOKEN')
+            if gh_token:
+                self.config['repo_token'] = gh_token
+                return
+            raise CoverallsException(
+                'Running on Github Actions but GITHUB_TOKEN is not set. '
+                'Add "env: GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}" to '
+                'your step config.')
```

Comment:
I think moving this block into `load_config_from_github()` would help us keep things more encapsulated by service.